xlsxwriter/worksheet.py

```diff
@@ -3710,17 +3767,20 @@ def _check_dimensions(self, row, col, ignore_row=False, ignore_col=False):
                 return -2
 
         if not ignore_row:
-            if self.dim_rowmin is None or row < self.dim_rowmin:
+            dim_rowmin = self.dim_rowmin
```

Comment:
Is this really a performance improvement. Can Python not optimise this itself. Probably not. Which is probably why pypy is so much faster.

Reply:
yes, there is cost associated with the dot operator, especially if what is dotted is a functional property.